packages/workspace/src/browser/workspace-service.ts

```diff
@@ -90,6 +90,7 @@ export class WorkspaceService implements FrontendApplicationContribution {
             // The same window has to be preserved too (instead of opening a new one), if the workspace root is not yet available and we are setting it for the first time.
             const preserveWindow = options ? options.preserveWindow : !(await this.root);
             await this.server.setRoot(rootUri);
+            this._root = valid;
```

Comment:
Maybe only do this when "preserveWindow" is set? Then I think we preserve the previous behavior.